sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-queue-retry.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,230 @@
+      await db.studentAssessment.save(currentAssessment);
+
+      // Retry job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await retryProcessor.enqueueAssessmentRetryOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+      // Assert item was added to the queue.
+      const queueData = {
+        assessmentId: updatedApplication.currentAssessment.id,
+      };
+      expect(startApplicationAssessmentQueueMock.add).not.toBeCalledWith(
+        queueData,
+      );
+    });
+
+    it("Should not queue an assessment cancellation with 'cancellation queued' status when it has this status for less than the time configured for the queue.", async () => {
+      // Arrange
+      const application = await saveFakeApplication(db.dataSource);
+      const currentAssessment = application.currentAssessment;
+      currentAssessment.studentAssessmentStatus =
+        StudentAssessmentStatus.CancellationQueued;
+
+      // Gets one hour less than the needed time for the assessment retry.
+      currentAssessment.updatedAt =
+        await getAmountOfHoursAssessmentRetryWithHourOffset(-1);
+      await db.studentAssessment.save(currentAssessment);
+
+      // Retry job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await retryProcessor.enqueueAssessmentRetryOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+      // Assert item was added to the queue.
+      const queueData = {
+        assessmentId: updatedApplication.currentAssessment.id,
+      };
+      expect(startApplicationAssessmentQueueMock.add).not.toBeCalledWith(
+        queueData,
+      );
+    });
+
+    /**
+     * Gets date calculated from the current date minus the amount of hours assessment retry plus an hour offset.
+     * @param hourOffset hour offset.
+     * @returns a date calculated from the current date minus the amount of hours assessment retry plus an hour offset.
+     */
+    const getAmountOfHoursAssessmentRetryWithHourOffset = async (
+      hourOffset: number,
+    ): Promise<Date> => {
+      const queueConfig: QueueConfiguration =
+        await queueService.queueConfigurationDetails(
+          QueueNames.AssessmentWorkflowQueueRetry,
+        );
+      return addHours(
+        -(
+          queueConfig.queueConfiguration.amountHoursAssessmentRetry + hourOffset
+        ),
+      );
+    };
+
+    /**
+     * Load application and assessment data needed to execute the assert operations.
+     * @param applicationId application id.
+     * @returns application with related assessments (currentAssessment and currentProcessingAssessment).
+     */
+    const loadApplicationDataForAssertions = (
```

Comment:
Can we review if this method is needed since apparently it is used to retrieve the assessment id that is already available?